File: from-1801-to-1900/1845-seat-reservation-manager/solution.go
type SeatManager struct {
    seats []bool
}


func Constructor(n int) SeatManager {
    seats := make([]bool, n)
    return SeatManager{seats: seats}
}


func (this *SeatManager) Reserve() int {
    for i := 0; i < len(this.seats); i ++ {
        if !this.seats[i] {
            this.seats[i] = true
            return i + 1
        }
    }
    return -1
}


func (this *SeatManager) Unreserve(seatNumber int)  {
    this.seats[seatNumber - 1] = false
}


/**
 * Your SeatManager object will be instantiated and called as such:
 * obj := Constructor(n);
 * param_1 := obj.Reserve();
 * obj.Unreserve(seatNumber);
 */

File: from-1801-to-1900/1845-seat-reservation-manager/solution.py
class SeatManager:

    def __init__(self, n: int):
        self.h = [i+1 for i in range(n)]
        heapq.heapify(self.h)

    def reserve(self) -> int:
        return heapq.heappop(self.h)

    def unreserve(self, seatNumber: int) -> None:
        heapq.heappush(self.h, seatNumber)


# Your SeatManager object will be instantiated and called as such:
# obj = SeatManager(n)
# param_1 = obj.reserve()
# obj.unreserve(seatNumber)